build/root/usr/local/bin/run

```diff
@@ -128,74 +138,41 @@ else
     echo "Using '$ARTIFACT_DIR' to store the test artifacts"
 fi
 
-prepare_cluster_for_gpu_operator() {
-    toolbox/cluster/capture_environment.sh
-
-    entitle
-
-    if ! toolbox/nfd/has_nfd_labels.sh; then
-        toolbox/nfd/deploy_from_operatorhub.sh
-    fi
-
-    if ! toolbox/nfd/has_gpu_nodes.sh; then
-        toolbox/cluster/scaleup.sh
-        toolbox/nfd/wait_gpu_nodes.sh
-    fi
-}
-
-validate_gpu_operator_deployment() {
-    trap toolbox/gpu-operator/capture_deployment_state.sh ERR
-
-    toolbox/gpu-operator/wait_deployment.sh
-    toolbox/gpu-operator/run_gpu_burn.sh
-    toolbox/gpu-operator/capture_deployment_state.sh
-}
+if [ $1 == "gpu-operator_test-master-branch" ]; then
+        exit 0
+        #target="gpu-operator"
+        #action="test-master-branch"
+        #shift 1
+elif [ $1 == "gpu-operator_test-operatorhub" ]; then
+        target="sro"
+        action="test-master-branch"
+        shift 1
+elif [ $1 == "gpu-operator_test-helm" ]; then
+        target="gpu-operator"
+        action="test-helm"
+        shift 1
+elif [ $1 == "gpu-operator_undeploy-operatorhub" ]; then
+        target="gpu-operator"
+        action="undeploy-operatorhub"
+        shift 1
+else
+        script_name="$0"
+        target="$1"
+        action="$2"
+        shift 2
+fi
 
 set -x
-case ${1:-} in
-    "gpu-operator_test-master-branch")
-        CI_IMAGE_GPU_COMMIT_CI_REPO="${2:-https://github.com/NVIDIA/gpu-operator.git}"
-        CI_IMAGE_GPU_COMMIT_CI_REF="${3:-master}"
-
-        CI_IMAGE_GPU_COMMIT_CI_IMAGE_UID="ci-image"
 
-        echo "Using Git repository ${CI_IMAGE_GPU_COMMIT_CI_REPO} with ref ${CI_IMAGE_GPU_COMMIT_CI_REF}"
-
-        prepare_cluster_for_gpu_operator
-        toolbox/gpu-operator/deploy_from_commit.sh "${CI_IMAGE_GPU_COMMIT_CI_REPO}" \
-                                                   "${CI_IMAGE_GPU_COMMIT_CI_REF}" \
-                                                   "${CI_IMAGE_GPU_COMMIT_CI_IMAGE_UID}"
-        validate_gpu_operator_deployment
-        exit 0
-        ;;
-    "gpu-operator_test-operatorhub")
-        OPERATOR_VERSION="${2:-}"
-        prepare_cluster_for_gpu_operator
-        toolbox/gpu-operator/deploy_from_operatorhub.sh ${OPERATOR_VERSION}
-        validate_gpu_operator_deployment
-        exit 0
+case ${target:-} in
+    "gpu-operator")
+        source /var/lib/ci-artifacts/entrypoints.d/gpu-operator-run-libs.sh
+        $action "$@"
```

Comment:
this cannot work, if `action = rm` and `"$@" = / -rf` , you're screwed ;-)
joke aside, I would prefer keeping `$action` in a switch-case statement, or any other controller mechanism, like something like that in Python:
```
cmd = {
"test_a": fct_a
"test_b": fct_b
}

cmd[target]()
```